include/supl_os_client.rst

```diff
@@ -1,30 +1,203 @@
+* Executing subsequent message transfers until the completion of the A-GPS session
+* Pushing data into the GPS module
+
+It is assumed that the GNSS socket is already available.
+
+The following Message Sequence Chart (MSC) describes the flow of communication in a SUPL session.
+
+.. _supl_session_image:
+
+.. figure:: /images/supl_msc.svg
+   :alt: SUPL Session msc
+
+The various steps in the communication session are described below:
+
+1. The application receives a request from the GPS to start a SUPL session.
+   This request message (``NRF_GNSS_AGPS_DATA_ID``) is sent periodically from the GPS and received on the GNSS socket.
+
+#. If the LTE modem is deactivated the user must enable the LTE modem.
+   When modem is enabled it initiates a TCP socket to connect to the SUPL server. In the SUPL session MSC above, the application opens a TCP socket to connect to the google SUPL server.
+#. Before starting the SUPL session the user must create a device ID and prepare the following necessary data needed for the session.
+
+   a. :cpp:member:`nrf_gnss_agps_data_frame_t` received from GPS in step 1
+   #. ``lte_params_t`` read from LTE modem
+   #. ``device_id`` made from IMSI number
+
+#. The session is initiated by calling the :cpp:func:`supl_client_session()` function.
+   The application does not return from this function until the SUPL session is finished or the :cpp:func:`supl_client_abort()` function is called.
+#. The :cpp:func:`supl_client_session()` function forms the entry point for the SUPL session.
+   There is no specific limit on the number of subsequent SUPL sessions that can be executed from this entry point.
+   See the documentation on the `SET-Initiated Non-Roaming Successful Case (Proxy mode)`_ for more information on the SUPL session.
+   The callback functions used for data transfer are listed below:
+
+    * SUPL Write (:cpp:func:`supl_write_t()`) : callback for sending outgoing data to the SUPL server
+    * SUPL Read (:cpp:func:`supl_read_t()`) : callback for receiving incoming data from the SUPL server
+
+#. The decoded SUPL data is sent to the GPS module using the AGPS Handler (:cpp:func:`agps_handler_t()`) callback function.
+#. After the application returns from the :cpp:func:`supl_client_session()` function, the TCP socket is no longer used by the SUPL client library and it can be closed.
+
+LPP2NRF
+=======
+
+This module converts SUPL structures into data structures defined in :file:`nrf_socket.h` used by the nRF GNSS socket.
+Pointers of the structures are forwarded to the AGPS Handler callback function.
+Pointers are valid till the return of the callback function, after which they fall out of scope.
+
+SUPL client OS integration library
+**********************************
+
+The SUPL client OS integration module integrates the SUPL client library into the nRF Connect SDK.
+This library can be used to receive A-GPS data from a SUPL server using the SUPL protocol.
```

Comment:
Should we move this sentence to line 9?

Reply:
Yes, it's out of place here, since we've already talked so much about the library. ;)